usecases/schema/add.go

```diff
@@ -160,8 +160,22 @@ func (m *Manager) addClass(ctx context.Context, class *models.Class,
 	}
 
 	if err := m.cluster.CommitWriteTransaction(ctx, tx); err != nil {
-		return nil, errors.Wrap(err, "commit cluster-wide transaction")
+		// Only log the commit error, but do not abort the changes locally. Once
```

Comment:
I think for the assumption to hold at least one node have to commit.
Following same example let say node 2 is also unavailable in this case the coordinator needs actually to abort the transaction. This issue becomes more clear in case of a two node cluster where the coordinator will always commit even if the partner node is unavailable.

One possible solution would be to change `CommitWriteTransaction()` to return the number of successes in addition to the error.  The code would look similar to the sudo code below
```
number_of_successes,err := m.cluster.CommitWriteTransaction(ctx, tx)
if err != nil{
	if number_of_successes == 0{ //abort if all nodes fail to commit 
		return nil, errors.Wrap(err, "commit cluster-wide transaction")
	}
	m.logger.WithError(err).Errorf("not every node was able to commit")
}
```

Reply:
The problem here is that we don't know _when_ the node failed to commit. It could also be that some nodes applied their changes already and only died between then and the response. Then the succes count would still be 0, but some nodes already applied the change. 

I think it's safer to treat any commit as a guaranteed commit. Once we have a WAL in the first phase, the changes would always be durable and the dead node could recover the missing change from the WAL on restart.

Reply:
Merging now because this is needed for a release that we want to roll out in a maintenance window – but happy to continue this conversation also after merging.